enhancements/baremetal/baremetal-keepalived-unicast.md

```diff
@@ -0,0 +1,184 @@
+of network policy. 
+
+Additionally, moving to unicast will also solve the possible collisions when two or more clusters are on the same broadcast/multicast domain. This use case described in more detail in `user story #2` below.
+
+In environments where there are lots of existing virtual routers managing Virtual IPs with multicast messaging or disallow the multicast VRRP traffic completely, it is necessary to configure Keepalived unicast communication  rather than making it an optional configuration. Making it default ensures a successful deployment, so It makes sense to make it the default and if we want to continue to support multicast VRRP, have it as an option.
+
+### Goals
+
+* Allow deployment in networks where multicast VRRP traffic is forbidden or non-functional.
+* No loss of functionality from the current multicast implementation.
+* Prevent VRRP ID collisions during deployment.
+* Moving existing clusters to unicast mode with minimal downtime.
+
+### Non-Goals
+
+* A switchable implementation. It does not need to be an installer choice.
+* Allow masters to be deployed on different subnets (not on the same L2 segment).
+
+## Proposal
+
+Change the way we configure and update Keepalived to manage the VIPs using unicast peer lists.
+
+### User Stories
+
+#### Story 1 - Allow deployment where networking doesn't permit multicast
+
+As an administrator, I want to deploy OpenShift on bare metal nodes with 
+installer-provisioned infrastructure (IPI) where the networking doesn't permit multicast.
+Some customers have restrictions on multicast network traffic (it impacts telco/edge customers since it’s not possible to use multicast between the regional and far-edge locations).
+The current deployment supports high availability for API and default ingress using multicast traffic for VRRP management of the VIP addresses.
+
+
+#### Story 2 - Allow multiple deployments on the same network
+
+For PoC and test deployments we might need to run multiple clusters on the same network (requiring a dedicated network is a heavy dependency in non-virtualized environments), while we’d still recommend isolated networks for production it would be desirable to enable deployment of multiple clusters on a single broadcast domain.
+Keepalived uses the `virtual_router_id` field to specify to which VRRP router the [instance](https://www.keepalived.org/doc/configuration_synopsis.html#vrrp-instance-definitions-synopsis) belongs,`virtual_router_id` range is 1 - 255.
+Each baremetal deployment includes two virtual routers:
+* API
+* Ingress
+
+Although the `virtual_router_id` values for both API and Ingress are calculated using [checksum function based on cluster name](https://github.com/openshift/baremetal-runtimecfg/blob/master/pkg/config/node.go#L162-#L171) we can still end up in a situation where virtual routers from different deployments or services share the same `virtual_router_id`.
+
+### Implementation Details/Notes/Constraints
```

Comment:
will it be useful to describe some recovery scenarios?

for example,
let's say we lose 2 out of 3 etcd, here we could loose access to kube-apiserver for getting information, how would the recovery process look like?

Reply:
I'm not familiar with recovery scenarios.
I'll clarify this point and in case it's relevant for Keepalived mode I'll add it to this enhancement.